src/main/scala/bubblewrap/HttpClient.scala

```diff
@@ -1,83 +1,76 @@
 package bubblewrap
 
 import java.net.URL
-import java.security.cert.X509Certificate
-import javax.net.ssl.{HostnameVerifier, SSLContext, SSLSession, X509TrustManager}
+import java.util.concurrent.TimeUnit
 
-import com.ning.http.client._
-import com.ning.http.client.cookie.Cookie
-import org.jboss.netty.handler.codec.http.HttpHeaders.Names._
-import org.jboss.netty.handler.codec.http.HttpHeaders.Values._
+import io.netty.handler.ssl.{SslContextBuilder, SslProvider}
+import io.netty.util.HashedWheelTimer
+import org.asynchttpclient.proxy.ProxyServer
+import org.asynchttpclient.{DefaultAsyncHttpClient, DefaultAsyncHttpClientConfig, Realm}
+import io.netty.handler.codec.http.HttpHeaders.Names._
+import io.netty.handler.codec.http.HttpHeaders.Values._
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory
+import org.asynchttpclient.cookie.Cookie
+import org.asynchttpclient.netty.NettyResponseFuture
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.collection.JavaConverters._
+import scala.util.Try
 
-class HttpClient(clientSettings:ClientSettings = new ClientSettings()) {
-  val lenientSSLContext: SSLContext = {
-    val context = SSLContext.getInstance("SSL")
-    context.init(null, Array(new X509TrustManager {
-      override def checkClientTrusted(chain:Array[X509Certificate], authType:String) {}
-      override def checkServerTrusted(chain:Array[X509Certificate], authType:String) {}
-      override def getAcceptedIssuers = null
-    }
-    ), null)
-    context
-  }
+class HttpClient(clientSettings: ClientSettings = ClientSettings()) {
+  val lenientSSLContext = SslContextBuilder.forClient().sslProvider(SslProvider.JDK).trustManager(InsecureTrustManagerFactory.INSTANCE).build()
 
-  val allHostsValid = new HostnameVerifier {
-    override def verify(s: String, sslSession: SSLSession): Boolean = true
-  }
-  
-  val client = new AsyncHttpClient(new AsyncHttpClientConfigBean()
-                                      .setConnectionTimeOut(clientSettings.connectionTimeout)
-                                      .setRequestTimeout(clientSettings.requestTimeout)
-                                      .setReadTimeout(clientSettings.readTimeout)
-                                      .setReadTimeout(clientSettings.socketTimeout)
-                                      .setAllowPoolingConnection(clientSettings.poolConnections)
-                                      .setAllowSslConnectionPool(clientSettings.poolConnections)
-                                      .setSslContext(lenientSSLContext)
-                                      .setAcceptAnyCertificate(true)
-                                      .setHostnameVerifier(allHostsValid)
-                                      .setMaxRequestRetry(clientSettings.retries)
-                                      .setFollowRedirect(false))
+  val timer = new HashedWheelTimer(100, TimeUnit.MILLISECONDS, 3072)
```

Comment:
@G1GC What do you think about making these values configurable, with these being the defaults? So if we want to change it we can directly do it where it's being used instead of making a release from here and then upgrading it.